File: LeetCode/Easy/0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.cpp
class KthLargest {
public:
    int k;
    priority_queue<int, vector<int>, greater<int>> minHeap;
    
    KthLargest(int k, vector<int>& nums) {
        this->k = k;
        for (int v : nums) add(v);
    }
    
    int add(int val) {
        if (minHeap.size() < k) minHeap.push(val);
        else if (minHeap.top() < val) {
            minHeap.pop();
            minHeap.push(val);
        }

        return minHeap.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: LeetCode/Easy/0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.java
class KthLargest {
    private int k;
    private PriorityQueue<Integer> minHeap;
    
    public KthLargest(int k, int[] nums) {
        this.k = k;
        minHeap = new PriorityQueue<>(k);

        for (int v : nums) add(v);
    }
    
    public int add(int val) {
        if (minHeap.size() < k) minHeap.offer(val);
        else if (minHeap.peek() < val){
            minHeap.poll();
            minHeap.offer(val);
        }
        
        return minHeap.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */